src/github.ts

```diff
@@ -188,41 +185,51 @@ export class GitHubConnection {
   async getOwnerRepos(owner: string): Promise<GitHubRepoData[]> {
     // Try to get the repo names assuming owner is an org.
     const allRepos: GitHubRepoData[] = [];
-    let pageRepos: GitHubRepoData[] = [];
-    const pageSize = 50;
-    let page = 1;
     let isOrg = true;
+    let hasNextPage = false;
+    let responseData;
+    let responseMeta;
 
-    do {
-      let responseData;
-      if (isOrg) {
-        try {
-          const response = await this._github.repos.getForOrg(
-              {org: owner, per_page: pageSize, page: page});
-          responseData = response.data;
-        } catch (e) {
-          // Owner is not an org? Continue as if owner is a user.
-          isOrg = false;
-        }
+    if (isOrg) {
+      try {
+        const response = await this._github.repos.getForOrg({org: owner});
+        responseData = response.data;
+        responseMeta = response.meta;
+      } catch (e) {
+        // Owner is not an org? Continue as if owner is a user.
+        isOrg = false;
+      }
+    }
+    if (!isOrg) {
+      try {
+        const response = await this._github.repos.getForUser({username: owner});
+        responseData = response.data;
+        responseMeta = response.meta;
+      } catch (e) {
+        // Owner is not an user, either? End and return any repos found.
+        responseData = [];
       }
-      if (!isOrg) {
+    }
+
+    while (hasNextPage || responseData.length > 0) {
+      responseData.filter((obj: any) => !obj.private)
+          .map(createGitHubRepoDataFromApi)
+          .forEach((repo: GitHubRepoData) => {
+            this.setCache(repo.fullName, repo);
+            allRepos.push(repo);
+          });
+
+      if (hasNextPage = !!this._github.hasNextPage(responseMeta)) {
         try {
-          const response = await this._github.repos.getForUser(
-              {username: owner, per_page: pageSize, page: page});
+          const response: any = await this._github.getNextPage(responseMeta);
```

Comment:
Wasn't sure why this line specifically was complaining about implicit type any, any thoughts?

Reply:
huh, strange. Different TS version?